phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/common/widgets/lists/ListWidgetEntry.java

```diff
@@ -105,7 +117,41 @@ public ListWidgetEntry(E item, Optional<URI> iconUri, URI defaultIconUri, String
     }
 
     public static ListWidgetEntry<ContainerDTO> create(ContainerDTO container) {
-        return new ListWidgetEntry<>(container, Optional.empty(), StaticMiniature.CONTAINER_MINIATURE,
+        Optional<URI> shortcutMiniature = Optional.empty();
+
+        final List<BufferedImage> miniatures = new ArrayList<>();
+        // do not use too many segments (cannot recognize the miniature if the segment is too small)
+        final int maxSegments = 4;
+        int currentSegment = 0;
+        for (ShortcutDTO shortcutDTO : container.getInstalledShortcuts()) {
+            if (currentSegment >= maxSegments) {
+                break;
+            }
+            try {
+                miniatures.add(ImageIO.read(shortcutDTO.getMiniature().toURL()));
+                currentSegment++;
+            } catch (IOException e) {
+                LOGGER.warn(String.format("Could not read miniature for shortcut \"%s\"", shortcutDTO.getName()), e);
+            }
+        }
+
+        BufferedImage segmentedMiniature = createSegmentedMiniature(miniatures);
+
+        if (segmentedMiniature != null) {
```

Comment:
Maybe move this in a new method:
```java
private Optional<URI> saveSegmentedBufferedImage(BufferedImage segmentedMiniature)
```
Or even integrate the method `createSegmentedMiniature` into this.